### ISSUE TITLE
Ability to save answers after lesson for later viewing

### ISSUE THREAD

**Issue by Midataur** [Midataur/mathsender#55]
Mr Finch has suggested that we add a way for teachers to save all the answers to view later. The idea is that this data can then be analyzed for things like common mistakes, which students are answering, what percentage of students got a questions correct, etc etc. He suggested this could be really good especially for setting online worksheets in tandem with question importing and seeing how well each student individually does, especially for year 10s who can be "hard to judge/wrangle". 

Suggested implementation:
* Export button similar to question exports
* Could we hijack the existing one? Idk
* Export as a file of some sort (csv or json probably)
* Viewing later should preserve latex rendering so could be good to have an analysis tool on the site
* This would probably be a page dedicated to answer analysis
* At this point we would probably need a proper navbar

**Comment by ehne** [comment]
hmm, now, this is a good idea, probably just export them as a csv with student names as rows, and question as columns?
also, we should probably make sure to get the wording correct on this one, so that you can't confuse saving the question set and downloading the current session data.

**Comment by Midataur** [comment]
Yeah that would be important. I like the analysis potential of csv but
there is the issue of it not rendering latex.

**Comment by ehne** [comment]
yeah, I'm not sure how to deal with that problem, maybe just somehow try and convert it to simplified latex kinda thing maybe? idk quite how to go about this.

**Comment by Midataur** [comment]
I think having an analysis page on our site would be the best way to go

**Comment by ehne** [comment]
hmm, whilst thats probably a nice idea, I reckon the teachers are going to want a spreadsheet of data.
I just had a thought tho: if we can somehow make it export to a google sheets doc, then we can render the latex as an image and use it as the headers? would that work? I just kinda want to give them a data sheet, because that's what I would expect from exporting answer data.

**Comment by ehne** [comment]
like, we should make a seperate api endpoint for the latex rendering bit maybe? I'm just kinda throwing out different ideas here.

**Comment by ehne** [comment]
or we could create ms excel docs?

**Comment by Midataur** [comment]
Eghh, it just seems janky at that point

**Comment by ehne** [comment]
hmm, I guess

**Comment by jmang00** [comment]
Hmm so google sheets (and excel) allows you to insert an image in a cell, maybe we could find an api or something to convert latex into an image. But then you'd have to auto generate a google sheet which might be hard.

**Comment by jmang00** [comment]
Google Apps Script actually seems promising.
https://zapier.com/learn/google-sheets/google-apps-script-tutorial/
https://developers.google.com/apps-script/reference/spreadsheet/spreadsheet

There's methods:
- insertImage(url, column, row)
- appendRow(rowContents)
- addEditors(emailAddresses)

**Comment by jmang00** [comment]
For latex to images:
https://dev.to/uetchy/math-api-latex-math-as-svg-image-m4p

**Comment by Midataur** [comment]
I mean sure, give it a good shot if you can

**Comment by jmang00** [comment]
yea might be fun. painful, but fun.

**Comment by ehne** [comment]
yeah, it's definitely something to look into.
